label_studio/ml_models/models.py

```diff
@@ -191,27 +191,37 @@ def delete_predictions(self):
         """
         predictions = Prediction.objects.filter(model_run=self.id)
         prediction_ids = [p.id for p in predictions]
+        logger.debug(f'Found predictions with IDs: {prediction_ids}')
+        
         # to delete all dependencies where predictions are foreign keys.
         Annotation.objects.filter(parent_prediction__in=prediction_ids).update(parent_prediction=None)
+        logger.debug(f'Annotations with parent predictions in {prediction_ids} have been updated to remove parent predictions')
+        
         try:
             from stats.models import PredictionStats
 
             prediction_stats_to_be_deleted = PredictionStats.objects.filter(prediction_to__in=prediction_ids)
+            logger.debug(f'PredictionStats to be deleted: {prediction_stats_to_be_deleted}')
             prediction_stats_to_be_deleted.delete()
         except Exception as e:
-            logger.info(f'PredictionStats model does not exist , exception:{e}')
+            logger.info(f'PredictionStats model does not exist, exception: {e}')
 
         # Delete failed predictions. Currently no other model references this, no fk relationships to remove
         failed_predictions = FailedPrediction.objects.filter(model_run=self.id)
         failed_predictions_ids = [p.id for p in failed_predictions]
+        logger.debug(f'Found failed predictions with IDs: {failed_predictions_ids}')
 
         # delete predictions meta
         PredictionMeta.objects.filter(prediction__in=prediction_ids).delete()
+        logger.debug(f'PredictionMeta for predictions {prediction_ids} have been deleted')
         PredictionMeta.objects.filter(failed_prediction__in=failed_predictions_ids).delete()
+        logger.debug(f'PredictionMeta for failed predictions {failed_predictions_ids} have been deleted')
 
         # remove predictions from db
-        predictions._raw_delete(predictions.db)
-        failed_predictions._raw_delete(failed_predictions.db)
+        predictions.delete()
```

Comment:
We were doing raw_delete for performance reasons, regular delete was taking a long time when we have lots of predictions. Deleting happens at the time of re-running a model version or updating a model version, and it was degrading UX so please test with a large project for example